lark/grammars/python.lark

```diff
@@ -225,17 +225,22 @@ _NEWLINE: ( /\r?\n[\t ]*/ | COMMENT )+
 NAME: /[a-zA-Z_]\w*/
 COMMENT: /#[^\n]*/
 
-STRING : /[ubf]?r?("(?!"").*?(?<!\\)(\\\\)*?"|'(?!'').*?(?<!\\)(\\\\)*?')/i
+STRING: /[ubf]?r?("(?!"").*?(?<!\\)(\\\\)*?"|'(?!'').*?(?<!\\)(\\\\)*?')/i
 LONG_STRING: /[ubf]?r?(""".*?(?<!\\)(\\\\)*?"""|'''.*?(?<!\\)(\\\\)*?''')/is
 
-DEC_NUMBER: /0|[1-9][\d_]*/i
-HEX_NUMBER.2: /0x[\da-f]*/i
-OCT_NUMBER.2: /0o[0-7]*/i
-BIN_NUMBER.2 : /0b[0-1]*/i
-FLOAT_NUMBER.2: /((\d+\.[\d_]*|\.[\d_]+)([Ee][-+]?\d+)?|\d+([Ee][-+]?\d+))/
-IMAG_NUMBER.2: /\d+[Jj]/ | FLOAT_NUMBER /[Jj]/
+DEC_NUMBER: "1".."9" _DEC_END | ("_"?  "0"                            )+ /(?![1-9])/
+_DEC_END:                       ("_"?  "0".."9"                       )*
+HEX_NUMBER.2: "0" ("x" | "X")   ("_"? ("0".."9" | "a".."f" | "A".."F"))+
+OCT_NUMBER.2: "0" ("o" | "O")   ("_"?  "0".."7"                       )+
+BIN_NUMBER.2: "0" ("b" | "B")   ("_"?  "0".."1"                       )+
```

Comment:
there's no lower/upper case (just normal one) for numbers(?), so the only thing we need to take from `/.../i` is in the hex, where we have `a-f` and in the prefixes `0something`